src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -557,10 +560,20 @@ private DRSAccessURL signAzureUrl(
   }
 
   private DRSAccessURL signGoogleUrl(
-      SnapshotCacheResult cachedSnapshot, String gsPath, AuthenticatedUserRequest authUser) {
+      SnapshotCacheResult cachedSnapshot,
+      String gsPath,
+      AuthenticatedUserRequest authUser,
+      String userProject) {
     Storage storage;
     if (cachedSnapshot.isSelfHosted) {
-      // In the case of a self-hosted dataset, use the dataset's service account to sign the url
+      // If a userProject is explicitly passed in, then use that to sign the url.
+      // Note: the expectation is that this is a Terra hosted bucket
```

Comment:
What happens if the bucket is not a Terra hosted bucket? Also, what if the user does not have access to the bucket?

Reply:
Ah good question.  in either of those cases, it will fail the request